demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureTypeService.php

```diff
@@ -363,16 +374,10 @@ public function addMissingRequestData(
 
                 return $return;
             })->toArray();
-            $params[$formName]['statementFormDefinition']['fieldDefinitions'] = $fieldDefinitions;
+            $statementFormDefinition['fieldDefinitions'] = $fieldDefinitions;
+            $request->request->set('statementFormDefinition', $statementFormDefinition);
         }
 
-        // Adds id to the form if missing
```

Comment:
Do i understand it correctly that this triggers the error? Otherwise the id is not set anymore. This was also handled in https://yaits.demos-deutschland.de/T21643 and initially in https://yaits.demos-deutschland.de/T19137

Reply:
No, the error was triggered by the removed line 374:
`$request->request->add([$formName => $params[$formName]]);`

Reply:
But shouldn't then the Id still be set?

Reply:
The issue (in line 374) was, that a key-value pair was add to the request.
This includes the $formName as key. The $formName was already deprecated (see BaseController.php line 345) and seems to me(!) not necessary for the logic anymore.
Without the $formName as key, the whole logic of setting the id does not make any sense anymore.
But maybe I have missed something and the $formName  is already necessary at some point?!